assets/js/modules/adsense/components/module/ModuleOverviewWidget/Overview.js

```diff
@@ -0,0 +1,115 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { calculateChange } from '../../../../../util';
+import DataBlock from '../../../../../components/DataBlock';
+
+const Overview = ( {
+	metrics,
+	currentRangeData,
+	previousRangeData,
+	selectedStats,
+	handleStatsSelection,
+} ) => {
+	const { totals, headers } = currentRangeData;
+	const { totals: previousTotals } = previousRangeData;
+
+	const dataBlocks = [
+		{
+			className: 'googlesitekit-data-block--page-rpm googlesitekit-data-block--button-1',
+			title: metrics[ headers[ 0 ].name ],
+			datapoint: totals[ 0 ],
+			datapointUnit: headers[ 0 ]?.currency,
+			change: calculateChange( previousTotals[ 0 ], totals[ 0 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 0,
+			handleStatsSelection,
+		},
+		{
+			className: 'googlesitekit-data-block--page-rpm googlesitekit-data-block--button-2',
+			title: metrics[ headers[ 1 ].name ],
+			datapoint: totals[ 1 ],
+			datapointUnit: headers[ 1 ]?.currency,
+			change: calculateChange( previousTotals[ 1 ], totals[ 1 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 1,
+			handleStatsSelection,
+		},
+		{
+			className: 'googlesitekit-data-block--impression googlesitekit-data-block--button-3',
+			title: metrics[ headers[ 2 ].name ],
+			datapoint: totals[ 2 ],
+			change: calculateChange( previousTotals[ 2 ], totals[ 2 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 2,
+			handleStatsSelection,
+		},
+		{
+			className: 'googlesitekit-data-block--impression googlesitekit-data-block--button-4',
+			title: metrics[ headers[ 3 ].name ],
+			datapoint: totals[ 3 ],
+			datapointUnit: '%',
+			change: calculateChange( previousTotals[ 3 ], totals[ 3 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 3,
+			handleStatsSelection,
+		},
+	];
+
+	return (
+		<div className="googlesitekit-adsense-performance-overview">
+			{ dataBlocks.map( ( block, i ) => (
+				<DataBlock
+					key={ i }
+					stat={ i }
+					className={ block.className }
+					title={ block.title }
+					datapoint={ block.datapoint }
+					datapointUnit={ block.datapointUnit }
+					change={ block.change }
+					changeDataUnit={ block.changeDataUnit }
+					context={ block.context }
+					selected={ block.selected }
+					handleStatSelection={ block.handleStatsSelection }
+				/>
+			) ) }
```

Comment:
> About using `map`, I prefer if we keep it like that, because it reduces the markup and everything is cleaner IMHO. Also, I think it'll be consistent about how we display the blocks on other modules `Overview` components as well.

I think there are pros/cons with both approaches, but I don't think we should iterate over a fixed list of props objects for a few reasons:
* It makes it easier to define one of the props wrong because the editor/IDE has no awareness of the component it is being used with. Since it's just a plain object, you can define anything and won't see it until it runs in the browser. Said another way, you lose the benefits of static analysis for prop names/types.
* There's actually a bit of a (potential) bug here in that each object above is intended to be a specific index (i.e. `selectedStats === 2`), but `stat` is being passed as the index from the `dataBlocks` array where everywhere else it is explicitly defined. If the order of the objects in the array changed, the stat each block refers to would no longer be accurate.

As for consistency, we have the inline approach in the [`AdSenseDashboardWidgetOverview`](https://github.com/google/site-kit-wp/blob/55c3280c1e61cbce8aa9360592c72a3fe452460b/assets/js/modules/adsense/components/dashboard/AdSenseDashboardWidgetOverview.js) component already. 

I think there's room to clean this up a bit more, but we can revisit that in a dedicated issue, perhaps after the legacy components are removed 👍

Reply:
👍  Reverted the changes to not use `map`